homeassistant/components/remote_calendar/calendar.py

```diff
@@ -0,0 +1,96 @@
+"""Calendar platform for a Remote Calendar."""
+
+from datetime import date, datetime, timedelta
+import logging
+
+from ical.event import Event
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util import dt as dt_util
+
+from . import RemoteCalendarConfigEntry
+from .const import CONF_CALENDAR_NAME
+from .coordinator import RemoteCalendarDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: RemoteCalendarConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the remote calendar platform."""
+    coordinator = entry.runtime_data
+    name = entry.data[CONF_CALENDAR_NAME]
+    entity = RemoteCalendarEntity(coordinator, name)
+    async_add_entities([entity])
+
+
+class RemoteCalendarEntity(
+    CoordinatorEntity[RemoteCalendarDataUpdateCoordinator], CalendarEntity
+):
+    """A calendar entity backed by a remote iCalendar url."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: RemoteCalendarDataUpdateCoordinator,
+        name: str,
+    ) -> None:
+        """Initialize RemoteCalendarEntity."""
+        super().__init__(coordinator)
+        self._attr_name = name
+        self._attr_unique_id = name
+
+    @property
+    def event(self) -> CalendarEvent | None:
+        """Return the next upcoming event."""
+        now = dt_util.now()
+        events = self.coordinator.data.timeline_tz(now.tzinfo).active_after(now)
+        if event := next(events, None):
+            return _get_calendar_event(event)
+        return None
+
+    async def async_get_events(
+        self, hass: HomeAssistant, start_date: datetime, end_date: datetime
+    ) -> list[CalendarEvent]:
+        """Get all events in a specific time frame."""
+        events = self.coordinator.data.timeline_tz(start_date.tzinfo).overlapping(
+            start_date,
+            end_date,
+        )
+        return [_get_calendar_event(event) for event in events]
+
+
+def _get_calendar_event(event: Event) -> CalendarEvent:
+    """Return a CalendarEvent from an API event."""
+    start: datetime | date
+    end: datetime | date
+    if isinstance(event.start, datetime) and isinstance(event.end, datetime):
```

Comment:
This logic for coercing dates in local calendar is here to cleanup a bug for how old calendar events were created https://github.com/home-assistant/core/pull/90970 Is there a specific reason this is all here or just copy/paste? If the latter, then remove and only include what is needed to comply with the `CalendarEvent` (e.g. ensure datetimes have a timezone)